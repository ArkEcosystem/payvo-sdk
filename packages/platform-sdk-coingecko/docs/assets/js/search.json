{"kinds":{"1":"Module","32":"Variable","128":"Class","512":"Constructor","1024":"Property","2048":"Method","2097152":"Object literal"},"rows":[{"id":0,"kind":1,"name":"\"transformers/historical-price-transformer\"","url":"modules/_transformers_historical_price_transformer_.html","classes":"tsd-kind-module"},{"id":1,"kind":128,"name":"HistoricalPriceTransformer","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"transformers/historical-price-transformer\""},{"id":2,"kind":512,"name":"constructor","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"transformers/historical-price-transformer\".HistoricalPriceTransformer"},{"id":3,"kind":1024,"name":"data","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html#data","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"transformers/historical-price-transformer\".HistoricalPriceTransformer"},{"id":4,"kind":2048,"name":"transform","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html#transform","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"transformers/historical-price-transformer\".HistoricalPriceTransformer"},{"id":5,"kind":1,"name":"\"transformers/historical-volume-transformer\"","url":"modules/_transformers_historical_volume_transformer_.html","classes":"tsd-kind-module"},{"id":6,"kind":128,"name":"HistoricalVolumeTransformer","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"transformers/historical-volume-transformer\""},{"id":7,"kind":512,"name":"constructor","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"transformers/historical-volume-transformer\".HistoricalVolumeTransformer"},{"id":8,"kind":1024,"name":"data","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html#data","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"transformers/historical-volume-transformer\".HistoricalVolumeTransformer"},{"id":9,"kind":2048,"name":"transform","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html#transform","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"transformers/historical-volume-transformer\".HistoricalVolumeTransformer"},{"id":10,"kind":1,"name":"\"transformers/market-transformer\"","url":"modules/_transformers_market_transformer_.html","classes":"tsd-kind-module"},{"id":11,"kind":128,"name":"MarketTransformer","url":"classes/_transformers_market_transformer_.markettransformer.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"transformers/market-transformer\""},{"id":12,"kind":512,"name":"constructor","url":"classes/_transformers_market_transformer_.markettransformer.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"transformers/market-transformer\".MarketTransformer"},{"id":13,"kind":1024,"name":"data","url":"classes/_transformers_market_transformer_.markettransformer.html#data","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"transformers/market-transformer\".MarketTransformer"},{"id":14,"kind":2048,"name":"transform","url":"classes/_transformers_market_transformer_.markettransformer.html#transform","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"transformers/market-transformer\".MarketTransformer"},{"id":15,"kind":1,"name":"\"index\"","url":"modules/_index_.html","classes":"tsd-kind-module"},{"id":16,"kind":128,"name":"PriceTracker","url":"classes/_index_.pricetracker.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"index\""},{"id":17,"kind":1024,"name":"tokenLookup","url":"classes/_index_.pricetracker.html#tokenlookup","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":18,"kind":1024,"name":"#httpClient","url":"classes/_index_.pricetracker.html#_httpclient","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":19,"kind":1024,"name":"#host","url":"classes/_index_.pricetracker.html#_host","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":20,"kind":512,"name":"constructor","url":"classes/_index_.pricetracker.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":21,"kind":2048,"name":"verifyToken","url":"classes/_index_.pricetracker.html#verifytoken","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":22,"kind":2048,"name":"marketData","url":"classes/_index_.pricetracker.html#marketdata","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":23,"kind":2048,"name":"historicalPrice","url":"classes/_index_.pricetracker.html#historicalprice","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":24,"kind":2048,"name":"historicalVolume","url":"classes/_index_.pricetracker.html#historicalvolume","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":25,"kind":2048,"name":"dailyAverage","url":"classes/_index_.pricetracker.html#dailyaverage","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":26,"kind":2048,"name":"getTokenId","url":"classes/_index_.pricetracker.html#gettokenid","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":27,"kind":2048,"name":"get","url":"classes/_index_.pricetracker.html#get","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":28,"kind":1,"name":"\"index.test\"","url":"modules/_index_test_.html","classes":"tsd-kind-module"},{"id":29,"kind":32,"name":"BASE_URL_COINGECKO","url":"modules/_index_test_.html#base_url_coingecko","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":30,"kind":32,"name":"token","url":"modules/_index_test_.html#token","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":31,"kind":32,"name":"currency","url":"modules/_index_test_.html#currency","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":32,"kind":32,"name":"subject","url":"modules/_index_test_.html#subject","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":33,"kind":1,"name":"\"transformers/historical-price-transformer.test\"","url":"modules/_transformers_historical_price_transformer_test_.html","classes":"tsd-kind-module"},{"id":34,"kind":32,"name":"stubResponse","url":"modules/_transformers_historical_price_transformer_test_.html#stubresponse","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\""},{"id":35,"kind":2097152,"name":"stubOptions","url":"modules/_transformers_historical_price_transformer_test_.html#stuboptions","classes":"tsd-kind-object-literal tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\""},{"id":36,"kind":32,"name":"type","url":"modules/_transformers_historical_price_transformer_test_.html#stuboptions.type","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\".stubOptions"},{"id":37,"kind":32,"name":"dateFormat","url":"modules/_transformers_historical_price_transformer_test_.html#stuboptions.dateformat","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\".stubOptions"},{"id":38,"kind":1,"name":"\"transformers/historical-volume-transformer.test\"","url":"modules/_transformers_historical_volume_transformer_test_.html","classes":"tsd-kind-module"},{"id":39,"kind":32,"name":"stubResponse","url":"modules/_transformers_historical_volume_transformer_test_.html#stubresponse","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\""},{"id":40,"kind":2097152,"name":"stubOptions","url":"modules/_transformers_historical_volume_transformer_test_.html#stuboptions","classes":"tsd-kind-object-literal tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\""},{"id":41,"kind":32,"name":"type","url":"modules/_transformers_historical_volume_transformer_test_.html#stuboptions.type","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\".stubOptions"},{"id":42,"kind":32,"name":"dateFormat","url":"modules/_transformers_historical_volume_transformer_test_.html#stuboptions.dateformat","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\".stubOptions"},{"id":43,"kind":1,"name":"\"transformers/market-transformer.test\"","url":"modules/_transformers_market_transformer_test_.html","classes":"tsd-kind-module"},{"id":44,"kind":32,"name":"stubResponse","url":"modules/_transformers_market_transformer_test_.html#stubresponse","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/market-transformer.test\""},{"id":45,"kind":2097152,"name":"stubOptions","url":"modules/_transformers_market_transformer_test_.html#stuboptions","classes":"tsd-kind-object-literal tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/market-transformer.test\""},{"id":46,"kind":32,"name":"type","url":"modules/_transformers_market_transformer_test_.html#stuboptions.type","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/market-transformer.test\".stubOptions"},{"id":47,"kind":32,"name":"dateFormat","url":"modules/_transformers_market_transformer_test_.html#stuboptions.dateformat","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/market-transformer.test\".stubOptions"}],"index":{"version":"2.3.9","fields":["name","parent"],"fieldVectors":[["name/0",[0,5.424,1,9.588,2,12.573]],["parent/0",[]],["name/1",[3,37.509]],["parent/1",[0,0.657,1,1.161,2,1.522]],["name/2",[4,25.689]],["parent/2",[0,0.657,1,1.161,5,1.988]],["name/3",[6,28.393]],["parent/3",[0,0.657,1,1.161,5,1.988]],["name/4",[7,28.393]],["parent/4",[0,0.657,1,1.161,5,1.988]],["name/5",[0,5.424,2,12.573,8,9.588]],["parent/5",[]],["name/6",[9,37.509]],["parent/6",[0,0.657,2,1.522,8,1.161]],["name/7",[4,25.689]],["parent/7",[0,0.657,8,1.161,10,1.988]],["name/8",[6,28.393]],["parent/8",[0,0.657,8,1.161,10,1.988]],["name/9",[7,28.393]],["parent/9",[0,0.657,8,1.161,10,1.988]],["name/10",[2,15.93,11,12.148]],["parent/10",[]],["name/11",[12,37.509]],["parent/11",[2,1.867,11,1.424]],["name/12",[4,25.689]],["parent/12",[11,1.424,13,2.439]],["name/13",[6,28.393]],["parent/13",[11,1.424,13,2.439]],["name/14",[7,28.393]],["parent/14",[11,1.424,13,2.439]],["name/15",[14,32.013]],["parent/15",[]],["name/16",[15,37.509]],["parent/16",[14,3.558]],["name/17",[16,37.509]],["parent/17",[17,1.733]],["name/18",[18,37.509]],["parent/18",[17,1.733]],["name/19",[19,37.509]],["parent/19",[17,1.733]],["name/20",[4,25.689]],["parent/20",[17,1.733]],["name/21",[20,37.509]],["parent/21",[17,1.733]],["name/22",[21,37.509]],["parent/22",[17,1.733]],["name/23",[22,37.509]],["parent/23",[17,1.733]],["name/24",[23,37.509]],["parent/24",[17,1.733]],["name/25",[24,37.509]],["parent/25",[17,1.733]],["name/26",[25,37.509]],["parent/26",[17,1.733]],["name/27",[26,37.509]],["parent/27",[17,1.733]],["name/28",[27,23.53]],["parent/28",[]],["name/29",[28,37.509]],["parent/29",[27,2.615]],["name/30",[29,37.509]],["parent/30",[27,2.615]],["name/31",[30,37.509]],["parent/31",[27,2.615]],["name/32",[31,37.509]],["parent/32",[27,2.615]],["name/33",[0,5.424,1,9.588,32,10.211]],["parent/33",[]],["name/34",[33,28.393]],["parent/34",[0,0.657,1,1.161,32,1.236]],["name/35",[34,28.393]],["parent/35",[0,0.657,1,1.161,32,1.236]],["name/36",[35,28.393]],["parent/36",[0,0.657,1,1.161,36,1.522]],["name/37",[37,28.393]],["parent/37",[0,0.657,1,1.161,36,1.522]],["name/38",[0,5.424,8,9.588,32,10.211]],["parent/38",[]],["name/39",[33,28.393]],["parent/39",[0,0.657,8,1.161,32,1.236]],["name/40",[34,28.393]],["parent/40",[0,0.657,8,1.161,32,1.236]],["name/41",[35,28.393]],["parent/41",[0,0.657,8,1.161,36,1.522]],["name/42",[37,28.393]],["parent/42",[0,0.657,8,1.161,36,1.522]],["name/43",[11,12.148,32,12.938]],["parent/43",[]],["name/44",[33,28.393]],["parent/44",[11,1.424,32,1.516]],["name/45",[34,28.393]],["parent/45",[11,1.424,32,1.516]],["name/46",[35,28.393]],["parent/46",[11,1.424,36,1.867]],["name/47",[37,28.393]],["parent/47",[11,1.424,36,1.867]]],"invertedIndex":[["base_url_coingecko",{"_index":28,"name":{"29":{}},"parent":{}}],["constructor",{"_index":4,"name":{"2":{},"7":{},"12":{},"20":{}},"parent":{}}],["currency",{"_index":30,"name":{"31":{}},"parent":{}}],["dailyaverage",{"_index":24,"name":{"25":{}},"parent":{}}],["data",{"_index":6,"name":{"3":{},"8":{},"13":{}},"parent":{}}],["dateformat",{"_index":37,"name":{"37":{},"42":{},"47":{}},"parent":{}}],["get",{"_index":26,"name":{"27":{}},"parent":{}}],["gettokenid",{"_index":25,"name":{"26":{}},"parent":{}}],["historicalprice",{"_index":22,"name":{"23":{}},"parent":{}}],["historicalpricetransformer",{"_index":3,"name":{"1":{}},"parent":{}}],["historicalvolume",{"_index":23,"name":{"24":{}},"parent":{}}],["historicalvolumetransformer",{"_index":9,"name":{"6":{}},"parent":{}}],["host",{"_index":19,"name":{"19":{}},"parent":{}}],["httpclient",{"_index":18,"name":{"18":{}},"parent":{}}],["index",{"_index":14,"name":{"15":{}},"parent":{"16":{}}}],["index\".pricetracker",{"_index":17,"name":{},"parent":{"17":{},"18":{},"19":{},"20":{},"21":{},"22":{},"23":{},"24":{},"25":{},"26":{},"27":{}}}],["index.test",{"_index":27,"name":{"28":{}},"parent":{"29":{},"30":{},"31":{},"32":{}}}],["marketdata",{"_index":21,"name":{"22":{}},"parent":{}}],["markettransformer",{"_index":12,"name":{"11":{}},"parent":{}}],["price",{"_index":1,"name":{"0":{},"33":{}},"parent":{"1":{},"2":{},"3":{},"4":{},"34":{},"35":{},"36":{},"37":{}}}],["pricetracker",{"_index":15,"name":{"16":{}},"parent":{}}],["stuboptions",{"_index":34,"name":{"35":{},"40":{},"45":{}},"parent":{}}],["stubresponse",{"_index":33,"name":{"34":{},"39":{},"44":{}},"parent":{}}],["subject",{"_index":31,"name":{"32":{}},"parent":{}}],["token",{"_index":29,"name":{"30":{}},"parent":{}}],["tokenlookup",{"_index":16,"name":{"17":{}},"parent":{}}],["transform",{"_index":7,"name":{"4":{},"9":{},"14":{}},"parent":{}}],["transformer",{"_index":2,"name":{"0":{},"5":{},"10":{}},"parent":{"1":{},"6":{},"11":{}}}],["transformer\".historicalpricetransformer",{"_index":5,"name":{},"parent":{"2":{},"3":{},"4":{}}}],["transformer\".historicalvolumetransformer",{"_index":10,"name":{},"parent":{"7":{},"8":{},"9":{}}}],["transformer\".markettransformer",{"_index":13,"name":{},"parent":{"12":{},"13":{},"14":{}}}],["transformer.test",{"_index":32,"name":{"33":{},"38":{},"43":{}},"parent":{"34":{},"35":{},"39":{},"40":{},"44":{},"45":{}}}],["transformer.test\".stuboptions",{"_index":36,"name":{},"parent":{"36":{},"37":{},"41":{},"42":{},"46":{},"47":{}}}],["transformers/historical",{"_index":0,"name":{"0":{},"5":{},"33":{},"38":{}},"parent":{"1":{},"2":{},"3":{},"4":{},"6":{},"7":{},"8":{},"9":{},"34":{},"35":{},"36":{},"37":{},"39":{},"40":{},"41":{},"42":{}}}],["transformers/market",{"_index":11,"name":{"10":{},"43":{}},"parent":{"11":{},"12":{},"13":{},"14":{},"44":{},"45":{},"46":{},"47":{}}}],["type",{"_index":35,"name":{"36":{},"41":{},"46":{}},"parent":{}}],["verifytoken",{"_index":20,"name":{"21":{}},"parent":{}}],["volume",{"_index":8,"name":{"5":{},"38":{}},"parent":{"6":{},"7":{},"8":{},"9":{},"39":{},"40":{},"41":{},"42":{}}}]],"pipeline":[]}}