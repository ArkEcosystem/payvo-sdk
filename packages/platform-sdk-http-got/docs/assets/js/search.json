{"kinds":{"1":"Module","32":"Variable","128":"Class","512":"Constructor","1024":"Property","2048":"Method","16777216":"Reference"},"rows":[{"id":0,"kind":1,"name":"\"request\"","url":"modules/_request_.html","classes":"tsd-kind-module"},{"id":1,"kind":128,"name":"Request","url":"classes/_request_.request.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"request\""},{"id":2,"kind":2048,"name":"send","url":"classes/_request_.request.html#send","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-overwrite tsd-is-protected","parent":"\"request\".Request"},{"id":3,"kind":1024,"name":"_bodyFormat","url":"classes/_request_.request.html#_bodyformat","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-protected","parent":"\"request\".Request"},{"id":4,"kind":1024,"name":"_options","url":"classes/_request_.request.html#_options","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-inherited tsd-is-protected","parent":"\"request\".Request"},{"id":5,"kind":512,"name":"constructor","url":"classes/_request_.request.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":6,"kind":2048,"name":"baseUrl","url":"classes/_request_.request.html#baseurl","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":7,"kind":2048,"name":"asJson","url":"classes/_request_.request.html#asjson","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":8,"kind":2048,"name":"asForm","url":"classes/_request_.request.html#asform","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":9,"kind":2048,"name":"asMultipart","url":"classes/_request_.request.html#asmultipart","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":10,"kind":2048,"name":"bodyFormat","url":"classes/_request_.request.html#bodyformat","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":11,"kind":2048,"name":"contentType","url":"classes/_request_.request.html#contenttype","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":12,"kind":2048,"name":"acceptJson","url":"classes/_request_.request.html#acceptjson","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":13,"kind":2048,"name":"accept","url":"classes/_request_.request.html#accept","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":14,"kind":2048,"name":"withHeaders","url":"classes/_request_.request.html#withheaders","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":15,"kind":2048,"name":"withCacheStore","url":"classes/_request_.request.html#withcachestore","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":16,"kind":2048,"name":"timeout","url":"classes/_request_.request.html#timeout","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":17,"kind":2048,"name":"retry","url":"classes/_request_.request.html#retry","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":18,"kind":2048,"name":"withOptions","url":"classes/_request_.request.html#withoptions","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":19,"kind":2048,"name":"get","url":"classes/_request_.request.html#get","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":20,"kind":2048,"name":"head","url":"classes/_request_.request.html#head","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":21,"kind":2048,"name":"post","url":"classes/_request_.request.html#post","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":22,"kind":2048,"name":"patch","url":"classes/_request_.request.html#patch","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":23,"kind":2048,"name":"put","url":"classes/_request_.request.html#put","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":24,"kind":2048,"name":"delete","url":"classes/_request_.request.html#delete","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-inherited","parent":"\"request\".Request"},{"id":25,"kind":1,"name":"\"index\"","url":"modules/_index_.html","classes":"tsd-kind-module"},{"id":26,"kind":16777216,"name":"Request","url":"modules/_index_.html#request","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":27,"kind":1,"name":"\"request.test\"","url":"modules/_request_test_.html","classes":"tsd-kind-module"},{"id":28,"kind":32,"name":"subject","url":"modules/_request_test_.html#subject","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"request.test\""}],"index":{"version":"2.3.9","fields":["name","parent"],"fieldVectors":[["name/0",[0,18.971]],["parent/0",[]],["name/1",[0,18.971]],["parent/1",[0,1.812]],["name/2",[1,29.957]],["parent/2",[2,0.233]],["name/3",[3,29.957]],["parent/3",[2,0.233]],["name/4",[4,29.957]],["parent/4",[2,0.233]],["name/5",[5,29.957]],["parent/5",[2,0.233]],["name/6",[6,29.957]],["parent/6",[2,0.233]],["name/7",[7,29.957]],["parent/7",[2,0.233]],["name/8",[8,29.957]],["parent/8",[2,0.233]],["name/9",[9,29.957]],["parent/9",[2,0.233]],["name/10",[10,29.957]],["parent/10",[2,0.233]],["name/11",[11,29.957]],["parent/11",[2,0.233]],["name/12",[12,29.957]],["parent/12",[2,0.233]],["name/13",[13,29.957]],["parent/13",[2,0.233]],["name/14",[14,29.957]],["parent/14",[2,0.233]],["name/15",[15,29.957]],["parent/15",[2,0.233]],["name/16",[16,29.957]],["parent/16",[2,0.233]],["name/17",[17,29.957]],["parent/17",[2,0.233]],["name/18",[18,29.957]],["parent/18",[2,0.233]],["name/19",[19,29.957]],["parent/19",[2,0.233]],["name/20",[20,29.957]],["parent/20",[2,0.233]],["name/21",[21,29.957]],["parent/21",[2,0.233]],["name/22",[22,29.957]],["parent/22",[2,0.233]],["name/23",[23,29.957]],["parent/23",[2,0.233]],["name/24",[24,29.957]],["parent/24",[2,0.233]],["name/25",[25,24.849]],["parent/25",[]],["name/26",[0,18.971]],["parent/26",[25,2.373]],["name/27",[26,24.849]],["parent/27",[]],["name/28",[27,29.957]],["parent/28",[26,2.373]]],"invertedIndex":[["_bodyformat",{"_index":3,"name":{"3":{}},"parent":{}}],["_options",{"_index":4,"name":{"4":{}},"parent":{}}],["accept",{"_index":13,"name":{"13":{}},"parent":{}}],["acceptjson",{"_index":12,"name":{"12":{}},"parent":{}}],["asform",{"_index":8,"name":{"8":{}},"parent":{}}],["asjson",{"_index":7,"name":{"7":{}},"parent":{}}],["asmultipart",{"_index":9,"name":{"9":{}},"parent":{}}],["baseurl",{"_index":6,"name":{"6":{}},"parent":{}}],["bodyformat",{"_index":10,"name":{"10":{}},"parent":{}}],["constructor",{"_index":5,"name":{"5":{}},"parent":{}}],["contenttype",{"_index":11,"name":{"11":{}},"parent":{}}],["delete",{"_index":24,"name":{"24":{}},"parent":{}}],["get",{"_index":19,"name":{"19":{}},"parent":{}}],["head",{"_index":20,"name":{"20":{}},"parent":{}}],["index",{"_index":25,"name":{"25":{}},"parent":{"26":{}}}],["patch",{"_index":22,"name":{"22":{}},"parent":{}}],["post",{"_index":21,"name":{"21":{}},"parent":{}}],["put",{"_index":23,"name":{"23":{}},"parent":{}}],["request",{"_index":0,"name":{"0":{},"1":{},"26":{}},"parent":{"1":{}}}],["request\".request",{"_index":2,"name":{},"parent":{"2":{},"3":{},"4":{},"5":{},"6":{},"7":{},"8":{},"9":{},"10":{},"11":{},"12":{},"13":{},"14":{},"15":{},"16":{},"17":{},"18":{},"19":{},"20":{},"21":{},"22":{},"23":{},"24":{}}}],["request.test",{"_index":26,"name":{"27":{}},"parent":{"28":{}}}],["retry",{"_index":17,"name":{"17":{}},"parent":{}}],["send",{"_index":1,"name":{"2":{}},"parent":{}}],["subject",{"_index":27,"name":{"28":{}},"parent":{}}],["timeout",{"_index":16,"name":{"16":{}},"parent":{}}],["withcachestore",{"_index":15,"name":{"15":{}},"parent":{}}],["withheaders",{"_index":14,"name":{"14":{}},"parent":{}}],["withoptions",{"_index":18,"name":{"18":{}},"parent":{}}]],"pipeline":[]}}