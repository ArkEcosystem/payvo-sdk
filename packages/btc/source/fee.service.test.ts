import "jest-extended";

import nock from "nock";

import { createService, requireModule } from "../test/mocking.js";
import { FeeService } from "./fee.service";

const matchSnapshot = (transaction): void =>
    expect({
        min: transaction.min.toString(),
        avg: transaction.avg.toString(),
        max: transaction.max.toString(),
        static: transaction.static.toString(),
    }).toMatchSnapshot();

afterEach(() => nock.cleanAll());

beforeAll(() => nock.disableNetConnect());

describe("FeeService", () => {
    it("should get the fees", async () => {
        nock("https://btc-test.payvo.com:443")
            .get("/api/fees")
            .reply(200, {
                data: {
                    min: 0.00001074,
                    avg: 0.00001074,
                    max: 0.00180617,
                },
            });

        const result = await (await createService(FeeService, "btc.testnet")).all();

        expect(result).toContainAllKeys([
            "transfer",
            "secondSignature",
            "delegateRegistration",
            "vote",
            "multiSignature",
            "ipfs",
            "multiPayment",
            "delegateResignation",
            "htlcLock",
            "htlcClaim",
            "htlcRefund",
        ]);

        matchSnapshot(result.transfer);
        matchSnapshot(result.secondSignature);
        matchSnapshot(result.delegateRegistration);
        matchSnapshot(result.vote);
        matchSnapshot(result.multiSignature);
        matchSnapshot(result.ipfs);
        matchSnapshot(result.multiPayment);
        matchSnapshot(result.delegateResignation);
        matchSnapshot(result.htlcLock);
        matchSnapshot(result.htlcClaim);
        matchSnapshot(result.htlcRefund);
    });
});
