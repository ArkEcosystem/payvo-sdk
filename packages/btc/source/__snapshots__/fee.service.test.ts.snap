// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeeService should get the fees 1`] = `
Object {
  "avg": "1074",
  "max": "180617",
  "min": "1074",
  "static": "180617",
}
`;

exports[`FeeService should get the fees 2`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 3`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 4`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 5`] = `
Object {
  "avg": "1074",
  "max": "180617",
  "min": "1074",
  "static": "180617",
}
`;

exports[`FeeService should get the fees 6`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 7`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 8`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 9`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 10`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;

exports[`FeeService should get the fees 11`] = `
Object {
  "avg": "0",
  "max": "0",
  "min": "0",
  "static": "0",
}
`;
